### PR TITLE
Add scheduleToFrequency formatting with tests

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
This PR adds logic to read a StructuredCalendar and return a human readable string. This is useful for Schedules not created in the UI since we did some magic in the comment field in the UI to make it a nice label.

Unit tests added.

### Screenshots (if applicable) 📸 
Screen Shot 2023-06-20 at 4 23 23 PM

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [x] Unit tests added

### CODE REVIEW

File: src/lib/utilities/schedule-frequency-formatting.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+// Examples of output
+// Wednesday at 06:30pm UTC
+// Sunday, Monday, Tuesday, Wednesday, Thursday, Friday at 12:00pm UTC
+// Every 1, 15, 31 of the month at 12:00pm UTC
+// Every 1, 15, 31 of January, March, April, June, July, September, December at 12:00pm UTC
+
+const isDailyForTheMonth = (calendar: StructuredCalendar) => {
+  return (
+    calendar.dayOfMonth?.length === 1 &&
+    calendar.dayOfMonth[0]?.start === 1 &&
```
Comment (laurakwhit) [inline src/lib/utilities/schedule-frequency-formatting.ts:13]: Looking at the protos it seems like all the properties are optional and can be `null` for  `StructuredCalendar`.  If so, we might want this instead
```suggestion
    calendar.dayOfMonth?.[0]?.start === 1 &&
```

Although, TypeScript seems fine with it so maybe I'm misinterpreting 🤔

File: src/lib/components/schedule/schedule-frequency.svelte [hunk 2]
```diff
@@ -13,7 +14,7 @@
 <div class="flex flex-col {$$props.class}">
   {#if calendar}
     <p data-testid="schedule-calendar-frequency">
-      {calendar?.comment ?? ''} UTC
+      {structuredCalendarToFrequency(calendar) || calendar?.comment || ''}
```
Comment (rossedfort) [inline src/lib/components/schedule/schedule-frequency.svelte:17]: Could we just move the `|| calendar?.comment || ''` logic to `structuredCalendarToFrequency` ?

**Review by rossedfort — APPROVED** [review]
lgtm, might consider consolidating more of that template logic into the util you wrote though

File: src/lib/components/schedule/schedule-frequency.svelte (hunk 2 above)
Comment (Alex-Tideman) [reply src/lib/components/schedule/schedule-frequency.svelte:17]: good call

File: src/lib/utilities/schedule-frequency-formatting.ts (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/utilities/schedule-frequency-formatting.ts:13]: I have the calendar.dayOfMonth?.length === 1 check before that line

Comment (laurakwhit) [reply src/lib/utilities/schedule-frequency-formatting.ts:13]: 🤦‍♀️ that you do.

**Pull request merged**